packages/snap-preact-components/src/components/Atoms/SearchHeader/SearchHeader.tsx

```diff
@@ -117,13 +128,17 @@ export const SearchHeader = observer((properties: SearchHeaderProps): JSX.Elemen
 });
 
 export interface SearchHeaderProps extends ComponentProps {
-	controller: SearchController;
+	controller?: SearchController;
+	queryStore?: SearchQueryStore;
+	paginationStore?: SearchPaginationStore;
+
 	titleText?: string | ((data: data) => string);
 	subTitleText?: string | ((data: data) => string);
```

Comment:
Since the subtitleText only appears when there are results, should this be `resultsText` and `resultsSubText`?

But perhaps the subtitleText should be shown in all three possible conditions (landing page, results, no results)?

Reply:
hmm good question, I could see it going either way...

Reply:
Hmmm... I thought about this a bit, and I think the subTitle could be useful in all situations... but also these props could get us into a weird situation where we are unable to control this component. This is very tricky to allow customization for...

What do you think of this?

```
landingPageTitle
resultsTitle
resultsCorrectedTitle
noResultsTitle
subTitle
```

Reply:
works for me!